ogr/factory.py

```diff
@@ -81,10 +81,15 @@ def get_project(
     :return: GitProject using the matching implementation
     """
     kls = get_service_class(url=url, service_mapping_update=service_mapping_update)
+    repo_url = parse_git_repo(url)
 
     if custom_instances:
         for service_inst in custom_instances:
-            if isinstance(service_inst, kls) and service_inst.instance_url in url:
+            si_repo_url = parse_git_repo(repo_url.get_instance_url())
```

Comment:
This is the second time we need to do it -- maybe a time to have a property for that in the service class.

(I'll create a good-first-issue.)